atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBoundStore.java

```diff
@@ -70,6 +76,7 @@
     @GuardedBy("this")
     private long currentLimit = -1;
 
+    private final long id;
```

Comment:
I'm wondering why this is a long over a UUID?  And also for the parse logic we could leverage the parsing build for our ValueTypes to ensure our serialization is functional.  IE we know the length of a UUID and then take the long off the end all using `ValueType.converstToJava(byte[], UUID.sizeOf())`.